pkg/regression/v2_volumes.go

```diff
@@ -0,0 +1,148 @@
+			Expect(volume.Wwn).ShouldNot(BeEmpty())
+
+			// Dump volumes
+			response, status, err := client.ShowVolumes(volname1)
+			Expect(err).To(BeNil())
+			Expect(status.ResponseTypeNumeric).To(Equal(storageapi.ApiSuccess))
+			ShowVolumes(logger, response)
+		})
+
+		It("created volume should exist", func() {
+
+			logger := klog.FromContext(tc.Config.Ctx)
+			exists, err := client.CheckVolumeExists(volname1, sizeValue)
+			logger.V(3).Info("CheckVolumeExists", "name", volname1, "exists", exists)
+			Expect(err).To(BeNil())
+			Expect(exists).To(Equal(true))
+		})
+
+		It("unknown volume should NOT exist", func() {
+
+			logger := klog.FromContext(tc.Config.Ctx)
+			exists, err := client.CheckVolumeExists("unknown-volume", sizeValue)
+			logger.V(3).Info("CheckVolumeExists", "name", volname1, "exists", exists)
+			Expect(err).To(BeNil())
+			Expect(exists).To(Equal(false))
+		})
+
+		It("created volume should have valid WWN", func() {
+
+			logger := klog.FromContext(tc.Config.Ctx)
+			wwn, err := client.GetVolumeWwn(volname1)
+			logger.V(3).Info("GetVolumeWwn", "name", volname1, "wwn", wwn)
+			Expect(err).To(BeNil())
+			Expect(wwn).ShouldNot(Equal(""))
+		})
+
+		It("should be able to publish/map volume", func() {
+
+			logger := klog.FromContext(tc.Config.Ctx)
+			lun, err := client.PublishVolume(volname1, []string{client.Initiator})
+			logger.V(3).Info("PublishVolume", "name", volname1, "lun", lun)
+			Expect(err).To(BeNil())
+			Expect(lun).ToNot(Equal(0))
+		})
+
+		It("should be able to unmap volume", func() {
+
+			logger := klog.FromContext(tc.Config.Ctx)
+			status, err := client.UnmapVolume(volname1, tc.Config.StorageController.Initiator)
+			logger.V(3).Info("UnmapVolume", "name", volname1, "mc-response", status.ResponseTypeNumeric)
+			Expect(err).To(BeNil())
+			Expect(status.ResponseTypeNumeric).To(Equal(storageapi.ApiSuccess))
+		})
+
+		It("should be able to expand the volume", func() {
+
+			logger := klog.FromContext(tc.Config.Ctx)
+			status, err := client.ExpandVolume(volname1, expandSize)
+			logger.V(3).Info("ExpandVolume", "name", volname1, "mc-response", status.ResponseTypeNumeric)
+			Expect(err).To(BeNil())
+			Expect(status.ResponseTypeNumeric).To(Equal(storageapi.ApiSuccess))
+
+			// Show volumes
+			response, status, err := client.ShowVolumes(volname1)
+			Expect(err).To(BeNil())
+			Expect(status.ResponseTypeNumeric).To(Equal(storageapi.ApiSuccess))
+			ShowVolumes(logger, response)
+		})
+
+		It("should be able to copy the volume", func() {
+
+			logger := klog.FromContext(tc.Config.Ctx)
+			status, err := client.CopyVolume(volname1, volname2, tc.Config.StorageController.Pool)
+			logger.V(3).Info("CopyVolume", "from", volname1, "to", volname2, "mc-response", status.ResponseTypeNumeric)
+			Expect(err).To(BeNil())
+			Expect(status.ResponseTypeNumeric).To(Equal(storageapi.ApiSuccess))
+
+			// Show volumes
+			response, status, err := client.ShowVolumes(volname2)
```

Comment:
Is the destination volume immediately present after a copy operation is initiated, or only once it has completed?

Reply:
My understanding is that it is immediately available, but will double check.